src/core/reference/include/openvino/reference/multiply.hpp

```diff
@@ -4,31 +4,44 @@
 
 #pragma once
 
+#include <algorithm>
 #include <cstddef>
 
-#include "openvino/core/shape.hpp"
-#include "openvino/op/util/attr_types.hpp"
 #include "openvino/reference/autobroadcast_binop.hpp"
 
 namespace ov {
 namespace reference {
+namespace func {
+
+template <class T>
+constexpr T multiply(const T a, const T b) {
+    return a * b;
```

Comment:
what if use `std::multiplies`?

Reply:
Is done  intentional as it make smaller binary size (will add comment about).

It looks like when using `autobroadcast_binop` and using functor instead function pointer will increase binary size much faster. I think can be related that `autobroadcast_binop` is complex template and function pointer is somehow better optimized and less code is generated.